Can't import module in Python

```
ImportError while importing test module '/tmp/python-jsonschema-rs.ZU1CWjaa/src/jsonschema_rs-0.40.1/crates/jsonschema-py/tests-py/test_suite.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.14/site-packages/_pytest/python.py:498: in importtestmodule
    mod = import_path(
/usr/lib/python3.14/site-packages/_pytest/pathlib.py:587: in import_path
    importlib.import_module(module_name)
/usr/lib/python3.14/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
<frozen importlib._bootstrap>:1398: in _gcd_import
    ???
<frozen importlib._bootstrap>:1371: in _find_and_load
    ???
<frozen importlib._bootstrap>:1342: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:938: in _load_unlocked
    ???
/usr/lib/python3.14/site-packages/_pytest/assertion/rewrite.py:186: in exec_module
    exec(co, module.__dict__)
crates/jsonschema-py/tests-py/test_suite.py:7: in <module>
    import jsonschema_rs
test-env/lib/python3.14/site-packages/jsonschema_rs/__init__.py:3: in <module>
    from .jsonschema_rs import (
E   ImportError: /tmp/python-jsonschema-rs.ZU1CWjaa/src/jsonschema_rs-0.40.1/test-env/lib/python3.14/site-packages/jsonschema_rs/jsonschema_rs.abi3.so: undefined symbol: aws_lc_0_37_0_EVP_PKEY_bits
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,7 @@
 ### Changed
 
 - Exclude internal `tracker` field from `ValidationError` Debug output.
+- Switch HTTP client TLS crypto backend from `aws-lc-rs` to `ring` to simplify building from source on some Linux distributions. [#957](https://github.com/Stranger6667/jsonschema/issues/957)
 
 ### Fixed
 

--- a/crates/jsonschema-py/CHANGELOG.md
+++ b/crates/jsonschema-py/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 ## [Unreleased]
 
+### Changed
+
+- Switch HTTP client TLS crypto backend from `aws-lc-rs` to `ring` to fix import errors when building from source on some Linux distributions. [#957](https://github.com/Stranger6667/jsonschema/issues/957)
+
 ### Fixed
 
 - `Draft4Validator` now correctly validates large Python integers outside the i64/u64 range (e.g., `-9223372036854775809`, `18446744073709551616`) as valid for `type: integer`.

--- a/crates/jsonschema/Cargo.toml
+++ b/crates/jsonschema/Cargo.toml
@@ -13,7 +13,7 @@ authors.workspace = true
 
 [features]
 default = ["resolve-http", "resolve-file"]
-resolve-http = ["reqwest"]
+resolve-http = ["reqwest", "rustls"]
 resolve-file = []
 resolve-async = [
     "referencing/retrieve-async",
@@ -25,10 +25,13 @@ resolve-async = [
 arbitrary-precision = ["serde_json/arbitrary_precision", "dep:num-bigint"]
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
-reqwest = { version = "0.13", features = [
+reqwest = { version = "0.13", default-features = false, features = [
   "blocking",
   "json",
+  "http2",
+  "rustls-no-provider",
 ], optional = true }
+rustls = { version = "0.23", default-features = false, features = ["ring", "std", "tls12"], optional = true }
 tokio = { version = "1.0", features = ["fs", "rt"], optional = true }
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]

--- a/crates/jsonschema/src/retriever.rs
+++ b/crates/jsonschema/src/retriever.rs
@@ -14,6 +14,11 @@ pub enum HttpRetrieverError {
         path: std::path::PathBuf,
         source: std::io::Error,
     },
+    /// Failed to parse certificate.
+    CertificateParse {
+        path: std::path::PathBuf,
+        source: reqwest::Error,
+    },
     /// Failed to build HTTP client.
     ClientBuild(reqwest::Error),
 }
@@ -29,6 +34,13 @@ impl std::fmt::Display for HttpRetrieverError {
                     path.display()
                 )
             }
+            Self::CertificateParse { path, source } => {
+                write!(
+                    f,
+                    "Failed to parse certificate '{}': {source}",
+                    path.display()
+                )
+            }
             Self::ClientBuild(e) => write!(f, "Failed to build HTTP client: {e}"),
         }
     }
@@ -39,22 +51,22 @@ impl std::error::Error for HttpRetrieverError {
     fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
         match self {
             Self::CertificateRead { source, .. } => Some(source),
-            Self::ClientBuild(e) => Some(e),
+            Self::CertificateParse { source, .. } | Self::ClientBuild(source) => Some(source),
         }
     }
 }
 
 /// Load a certificate from a PEM file.
-/// Note: `reqwest::Certificate::from_pem()` is lenient and rarely fails.
-/// Invalid certificates are caught later during `ClientBuilder::build()`.
 #[cfg(all(feature = "resolve-http", not(target_arch = "wasm32")))]
 fn load_certificate(path: &std::path::Path) -> Result<reqwest::Certificate, HttpRetrieverError> {
     let cert_data = std::fs::read(path).map_err(|e| HttpRetrieverError::CertificateRead {
         path: path.to_path_buf(),
         source: e,
     })?;
-    // from_pem is lenient; invalid certs fail during builder.build()
-    Ok(reqwest::Certificate::from_pem(&cert_data).expect("from_pem should not fail"))
+    reqwest::Certificate::from_pem(&cert_data).map_err(|e| HttpRetrieverError::CertificateParse {
+        path: path.to_path_buf(),
+        source: e,
+    })
 }
 
 /// Configure an HTTP client builder with the given options.
@@ -115,6 +127,8 @@ impl HttpRetriever {
     /// - The certificate is not valid PEM
     /// - The HTTP client cannot be built
     pub fn new(options: &HttpOptions) -> Result<Self, HttpRetrieverError> {
+        // Install ring as the default TLS crypto provider (only needed once)
+        let _ = rustls::crypto::ring::default_provider().install_default();
         let builder = configure_http_client!(reqwest::blocking::Client::builder(), options);
         Ok(Self {
             client: builder.build().map_err(HttpRetrieverError::ClientBuild)?,
@@ -187,6 +201,8 @@ impl AsyncHttpRetriever {
     /// - The certificate is not valid PEM
     /// - The HTTP client cannot be built
     pub fn new(options: &HttpOptions) -> Result<Self, HttpRetrieverError> {
+        // Install ring as the default TLS crypto provider (only needed once)
+        let _ = rustls::crypto::ring::default_provider().install_default();
         let builder = configure_http_client!(reqwest::Client::builder(), options);
         Ok(Self {
             client: builder.build().map_err(HttpRetrieverError::ClientBuild)?,
@@ -256,6 +272,8 @@ impl Retrieve for DefaultRetriever {
             "http" | "https" => {
                 #[cfg(all(feature = "resolve-http", not(target_arch = "wasm32")))]
                 {
+                    // Install ring as the default TLS crypto provider (only needed once)
+                    let _ = rustls::crypto::ring::default_provider().install_default();
                     Ok(reqwest::blocking::get(uri.as_str())?.json()?)
                 }
                 #[cfg(all(feature = "resolve-http", target_arch = "wasm32"))]
@@ -311,7 +329,13 @@ impl referencing::AsyncRetrieve for DefaultRetriever {
         #[cfg(not(all(target_arch = "wasm32", target_os = "unknown")))]
         match uri.scheme().as_str() {
             "http" | "https" => {
-                #[cfg(feature = "resolve-http")]
+                #[cfg(all(feature = "resolve-http", not(target_arch = "wasm32")))]
+                {
+                    // Install ring as the default TLS crypto provider (only needed once)
+                    let _ = rustls::crypto::ring::default_provider().install_default();
+                    Ok(reqwest::get(uri.as_str()).await?.json().await?)
+                }
+                #[cfg(all(feature = "resolve-http", target_arch = "wasm32"))]
                 {
                     Ok(reqwest::get(uri.as_str()).await?.json().await?)
                 }
@@ -641,10 +665,9 @@ mod tests {
 
     #[test]
     #[cfg(all(feature = "resolve-http", not(target_arch = "wasm32")))]
-    fn test_http_retriever_error_client_build_display() {
+    fn test_http_retriever_error_invalid_certificate() {
         use std::io::Write;
 
-        // Invalid PEM content that passes from_pem() but fails during client build
         let mut temp = tempfile::NamedTempFile::new().unwrap();
         temp.write_all(b"-----BEGIN CERTIFICATE-----\ninvalid\n-----END CERTIFICATE-----")
             .unwrap();

